zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -1005,42 +1000,47 @@ private static List<String> applyUserSortField(
     return validationErrors;
   }
 
-  private static List<VariableValueFilter> toVariableValueFilters(
+  private static List<VariableValueFilter> toVariableValueFiltersForUserTask(
       final List<UserTaskVariableFilterRequest> filters) {
-
     if (filters != null && !filters.isEmpty()) {
-      return filters.stream()
-          .map(
-              filter -> {
-                final var builder = new VariableValueFilter.Builder().name(filter.getName());
-                if (filter.getValue() != null) {
-                  builder.valueOperation(UntypedOperation.of(eq(filter.getValue())));
-                }
-                return builder.build();
-              })
-          .toList();
+      final List<VariableValueFilter> variableValueFilters = new ArrayList<>();
+      filters.forEach(
+          filter ->
+              variableValueFilters.addAll(
+                  toVariableValueFilters(filter.getName(), filter.getValue())));
+      return variableValueFilters;
     }
-
     return null;
   }
 
   private static List<VariableValueFilter> toVariableValueFiltersForProcessInstance(
       final List<ProcessInstanceVariableFilterRequest> filters) {
-
     if (filters != null && !filters.isEmpty()) {
-      return filters.stream()
-          .map(
-              filter ->
-                  new VariableValueFilter.Builder()
-                      .name(filter.getName())
-                      .valueOperation(UntypedOperation.of(eq(filter.getValue())))
-                      .build())
-          .toList();
+      final List<VariableValueFilter> variableValueFilters = new ArrayList<>();
+      filters.forEach(
+          filter ->
+              variableValueFilters.addAll(
+                  toVariableValueFilters(filter.getName(), filter.getValue())));
+      return variableValueFilters;
     }
-
     return null;
   }
 
+  private static List<VariableValueFilter> toVariableValueFilters(
+      String name, StringFilterProperty value) {
+    Objects.requireNonNull(value);
+    final List<VariableValueFilter> variableValueFilters = new ArrayList<>();
+    final List<Operation<String>> operations = mapToOperations(String.class).apply(value);
+    for (Operation<String> operation : operations) {
+      final UntypedOperation untypedOperation = UntypedOperation.of(operation);
+      final VariableValueFilter variableValueFilter =
+          new VariableValueFilter.Builder().name(name).valueOperation(untypedOperation).build();
+      variableValueFilters.add(variableValueFilter);
+    }
```

Comment:
💭 This looks misplaced here. We shouldn't have to take care of this here in the REST API, I believe. The variable value filter object should take a `name` and a list of operations for the `value` property. Breaking this down into multiple `name` and `value`-operation pairs should happen in the search layer. @danielkelemen, maybe you can advise here a bit as you know that part of the code better.